packages/core/src/api/parsers/handleFileInsertion.ts

```diff
@@ -72,9 +83,23 @@ export async function handleFileInsertion<
     let fileBlockType = "file";
     for (const fileBlockConfig of fileBlockConfigs) {
       for (const mimeType of fileBlockConfig.fileBlockAcceptMimeTypes || []) {
-        if (checkMIMETypesMatch(items[i].type, mimeType)) {
-          fileBlockType = fileBlockConfig.type;
-          break;
+        const isFileExtension = mimeType.startsWith(".");
```

Comment:
Let's change `const mimeType` to `const mimeTypeOrExtension` (technically, mimetype is different from a file extension). Maybe we should also change `fileBlockAcceptMimeTypes` to `fileBlockAccept`